### PR TITLE
Added documentation and example of W0416 Shadowed Import

### PR DESCRIPTION
**Pull request by sarahsonder**
## Motivation and Context
Since Shadowed Import is a new Pylint checker that was recently released and incorporated into PythonTA, this pull request was required to keep the PythonTA docs and examples up to date.

## Your Changes
The changes made in this pull request include adding documentation for checker Shadowed Import (W0416) and adding an example of python code that will trigger this error.

**Description**:

**Type of change** (select all that apply):

- [x] Documentation update (change that modifies or updates documentation only)

## Testing
I ran the example through Pylint and verified that the correct error was being raised. I also proofread my changes to the docs to check for grammatical errors.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls.

### CODE REVIEW

File: docs/checkers/index.md [hunk 1]
```diff
@@ -1151,6 +1151,16 @@ would like to import. For example, if we have a Python file named `math.py`, cal
 from within that file (or from within _any_ Python file in the same directory) will import _
 our_ `math.py` file, and not the [`math` module] from the standard library.
 
+(W0416)=
+
+### Shadowed import (W0416)
+
+This error occurs when a module is aliased with a name that has already been used by a previous import.
```
Comment (david-yz-liu) [inline docs/checkers/index.md:1158]: please reword "when a module is aliased with a name" to "when a module is imported with an aliased name"

Comment (david-yz-liu) [reply docs/checkers/index.md:1158]: Also add a second sentence: "This prevents the original module from ever being used later in your code."

**Comment by sarahsonder** [comment]
I think I accidentally made two commits when I only intended to make one, and I am not sure how to remove the commit. The correct changes should be in the first commit. Apologies!

**Review by david-yz-liu — COMMENTED** [review]
@sarahsonder I left one comment. It looks like the second commit you made was empty (you can see this under the "Commits" tab), so while I'm not sure why this happened, I don't think you need to worry about it. In general don't worry about "removing" commits when you're first starting out with git, just double-check the "Files changed" diff carefully and as long as it looks okay, that should be fine. :)

**Comment by sarahsonder** [comment]
Hi Prof. Liu, I have fixed the grammar error, but for some reason I have again made commits without meaning to. As you mentioned in your comment, I checked "git status" and "git diff" and nothing appeared out of the ordinary when I was making my commits. However, every time I tried pushing my commits, I got a message saying "Please enter a commit message to explain why this merge was necessary". My commits also kept getting rejected, which I fixed by pulling directly from my remote branch (in addition to pulling from the master branch before making my commits). Hopefully this error won't happen with future branches, but is there any way I can prevent this from happening again? Thank you!

**Comment by david-yz-liu** [comment]
@sarahsonder see https://stackoverflow.com/questions/19085807/please-enter-a-commit-message-to-explain-why-this-merge-is-necessary-especially

**Review by david-yz-liu — APPROVED** [review]

**Pull request merged**